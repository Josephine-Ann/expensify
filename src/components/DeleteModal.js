import React from 'react';
import Modal from 'react-modal';

const DeleteModal = (props) => (
    <Modal 
    isOpen={!!props.modalOpen}
    contentLabel="deletedOption"
    onRequestClose={props.clearStateCloseModal}
    onRemove={props.onRemove}
    >
       <h3>Are you sure you want to delete this?</h3>
       <button onClick={props.onRemove}>Yes!</button>
       <button onClick={props.clearStateCloseModal}>Noooo!</button>
    </Modal>
);

export default DeleteModal;
