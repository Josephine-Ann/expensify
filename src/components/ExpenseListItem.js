import React from 'react'; 
import { connect } from 'react-redux';
import { removeExpense } from '../actions/expenses';

const ExpenseListItem = ({ dispatch, description, amount, createdAt, id }) => (
    <div>
    <p>{description}</p>
    <p>{amount}</p>
    <p>{createdAt}</p>
    <button onClick={() => {
        dispatch(removeExpense(id));
    }}>Remove</button>
    </div>
);

export default connect()(ExpenseListItem);

// import React from 'react';
// import { connect } from 'react-redux';
// import { setTextFilter } from '../actions/filters';

// const ExpenseListFilters = (props) => (
//     <div>
//     <input type="text" value={props.filters.text} onChange={(e) => {
//         props.dispatch(setTextFilter(e.target.value));
//         console.log(e.target.value)
//     }}/>
//     </div>
// );

// const mapStateToProps = (state) => {
//     return {
//         filters: state.filters
//     }
// }

// export default connect(mapStateToProps)(ExpenseListFilters);
