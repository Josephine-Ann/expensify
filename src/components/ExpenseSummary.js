import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import expensesTotal from '../selectors/expenses-total'; 


export const ExpenseSummary = ({ expensesTotal, amountExpenses }) => (
    <div> 
    { amountExpenses !== 1 ? 
        <p>Your {amountExpenses} expenses come to {numeral(expensesTotal/ 100).format('$0,0.00')}</p> 
        : 
        <p>Your 1 expense comes to {numeral(expensesTotal/ 100).format('$0,0.00')}</p>}
    </div>
);

const mapStateToProps = (state) => {
    return {
        expensesTotal: expensesTotal(state.expenses),
        amountExpenses: state.expenses.length
    };
};

export default connect(mapStateToProps)(ExpenseSummary);
