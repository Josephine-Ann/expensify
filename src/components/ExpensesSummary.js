// MINE
// import React from 'react';
// import { connect } from 'react-redux';
// import numeral from 'numeral';
// import expensesTotal from '../selectors/expenses-total'; 


// export const ExpenseSummary = ({ expensesTotal, amountExpenses }) => (
//     <div> 
//     { amountExpenses !== 1 ? 
//         <p>Your {amountExpenses} expenses come to {numeral(expensesTotal/ 100).format('$0,0.00')}</p> 
//         : 
//         <p>Your 1 expense comes to {numeral(expensesTotal/ 100).format('$0,0.00')}</p>}
//     </div>
// );

// const mapStateToProps = (state) => {
//     return {
//         expensesTotal: expensesTotal(state.expenses),
//         amountExpenses: state.expenses.length
//     };
// };

// export default connect(mapStateToProps)(ExpenseSummary);
import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
  const expenseWord = expenseCount === 1 ? 'expense' : 'expenses' ;
  const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');
  
  return (
    <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title">Viewing <span>{expenseCount}</span> {expenseWord} totalling <span>{formattedExpensesTotal}</span></h1>
        <div className="page-header__actions">
          <Link className="button" to="/create">Add Expense</Link>
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);

  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);