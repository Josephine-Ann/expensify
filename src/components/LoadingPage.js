import React from 'react';

const LoadingPage = () => (
    <div className="loader">
        <img className="loader__image" src="/dist/images/loadere.gif"/>
    </div>   
);

export default LoadingPage;