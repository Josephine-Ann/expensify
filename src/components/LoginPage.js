import React from 'react';
import { connect } from 'react-redux';
import { startLoginGoogle, startLoginGithub } from '../actions/auth';


export const LoginPage = ({ startLoginGoogle, startLoginGithub }) => (
    <div className="box-layout">
    <div className="box-layout__box">
    <h1 className="box-layout__title">Expensify</h1>
    <p>It's time to get your expenses under control...</p>
    <button className="button" onClick={startLoginGoogle}> Login with Google </button>
    <button className="button" onClick={startLoginGithub}>Login with Github</button>
    </div>
    </div>
    );

const mapDispatchToProps = (dispatch) => ({
    startLoginGoogle: () => dispatch(startLoginGoogle()),
    startLoginGithub: () => dispatch(startLoginGithub())
}); 

export default connect(undefined, mapDispatchToProps)(LoginPage);