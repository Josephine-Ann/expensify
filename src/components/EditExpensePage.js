import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';
import DeleteModal from './DeleteModal'

export class EditExpensePage extends React.Component {
    state = {
        deletedOption: false
    };
    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    };
    onRemove = () => {
        this.props.startRemoveExpense({ id: this.props.expense.id });
        this.props.history.push('/');
    }
    onSelect = () => {
        this.setState({ deletedOption: true })
    }
    clearStateCloseModal = () => {
        this.setState(() => ({ deletedOption: null  }));
      }
    render() {
        return (
            <div>
                <div className="content-container">
                    <div className="page-header">
                        <h1 className="page-header__title">Edit Expense</h1>
                    </div>
                    <ExpenseForm 
                    expense={this.props.expense}
                    onSubmit={this.onSubmit}
                    />
                    <button className="button button--secondary" onClick={this.onSelect}>Remove expense</button>
                    <DeleteModal
                    modalOpen={this.state.deletedOption}
                    clearStateCloseModal={this.clearStateCloseModal}
                    onRemove={this.onRemove}
                    />
                </div>
            </div>
        );
    }
}
const mapStateToProps = (state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id )
});

const mapDispatchToProps = (dispatch, props) => ({
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
     startRemoveExpense: (id) => dispatch(startRemoveExpense(id))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);


 // yarn add react-modal@2.2.2
// make a new component 
// import Modal from 'react-modal';
// default arrow function with props 
// <Modal isOpen={!!props.selectedOption} (will resolve to true/false) contentLabel="Selected Option ">
// </Modal>
// above layer, track selectedOption as undefined
// Pass it down to option modal in render of that layer,
// handleDeleteOption = (optionToRemove) => {
// selectedOption={this.state.selectedOption}
//this.setState((prevState) => ({
//    options: prevState.options.filter((option) => optionToRemove !== option)
// }))}