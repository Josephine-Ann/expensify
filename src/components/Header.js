import React from 'react';
import { NavLink } from 'react-router-dom';
import ExpenseSummary from './ExpenseSummary';


const Header = () => (
    <header>
        <h1>Expensify</h1>
        <ExpenseSummary/>
        <NavLink activeClassName="isActive" exact={true} to="/">Dashboard</NavLink>
        <NavLink activeClassName="isActive" to="/create">Create an expense</NavLink>
        <NavLink activeClassName="isActive" to="/help">Help</NavLink>
    </header>
);

export default Header;

