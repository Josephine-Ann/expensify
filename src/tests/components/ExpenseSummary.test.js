import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';

test('should render ExpenseSummary correctly with multiple expenses', () => {
    const wrapper = shallow(<ExpenseSummary expenseTotal={3334455} amountExpenses={3}/>);
    expect(wrapper).toMatchSnapshot();
});

// test('should render ExpenseSummary correctly with no expenses', () => {
//     const wrapper = shallow(<ExpenseSummary expenses={[]}/>);
//     expect(wrapper).toMatchSnapshot();
// });

