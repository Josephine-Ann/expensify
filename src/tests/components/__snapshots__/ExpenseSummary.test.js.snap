// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary correctly with multiple expenses 1`] = `
<div>
  <p>
    Your 
    3
     expenses come to 
    $0.00
  </p>
</div>
`;
